snap/get_version

```diff
@@ -0,0 +1,114 @@
+# For instance:
+# * 0.1 -> 0.1
+# * msentraid-0.1 -> 0.1 on msentraid branch.
+
+# 1. If current commit is tagged, the version is directly the tag name.
+# 2. If current commit is not tagged, the version is:
+#    a. <last tag name on current branch>+<commit_sha> for main branch.
+#    b. <last tag name on current branch (not in main)>+<commit_sha>.<last_commit_merged_from_main> for other branches.
+#
+# Any of those version will be annoted with +dirty if there are local changes.
+
+# set_version will markup the version in the snapcraft.yaml file after amending it with a dirty markup if necessary.
+# $1: version: the version to set.
+set_version() {
+    version="${1}"
+
+    version=$(annotate_with_dirty "${version}")
+    craftctl set version="${version}"
+}
+
+# annotate_with_dirty may amend the version with a dirty markup if there are local changes.
+# $1: version: the version to annotate.
+annotate_with_dirty() {
+    version="${1}"
+
+    # check if current tree content is dirty.
+    if [ -n "$(git -C ${SNAPCRAFT_PART_SRC} status --porcelain)" ]; then
+        version="${version}+dirty"
+    fi
+
+    echo "${version}"
+}
+
+# last_tag_on_branch returns the last tag on that branch that is not in another provided branch.
+# $1: origin_branch: branch we start listing from.
+# $2: other_branch: the branch where we don’t want to tags to be reachable from.
+# If we don’t have another branch to compare with, it will just return the most recent tag.
+last_tag_on_branch() {
+    origin_branch="${1}"
+    other_branch="${2}"
+
+    all_tags_on_origin_branch=$(git -C ${SNAPCRAFT_PART_SRC} tag --sort=-committerdate --merged "${origin_branch}")
+    for tag in ${all_tags_on_origin_branch}; do
+        # Just return first commit if there is no branch to compare against the main branch.
+        if [ -z "${other_branch}" ]; then
+            tag=$(strip_branch_tag_prefix "${tag}" "${origin_branch}")
+            echo ${tag}
+            return
+        fi
+
+        # Skip the tags that are reachable from the other branch.
+        if git -C ${SNAPCRAFT_PART_SRC} merge-base --is-ancestor $(git -C ${SNAPCRAFT_PART_SRC} rev-list -n 1 "${tag}") "${other_branch}"; then
+            continue
+        fi
+
+        tag=$(strip_branch_tag_prefix "${tag}" "${origin_branch}")
+        echo "${tag}"
+        return
+    done
+
+    echo "notag"
+}
+
+# strip_branch_tag_prefix will remove the branch name prefix from the tag name.
+# $1: tag: the tag name to strip the prefix from.
+# $2: current_branch: the branch name to strip from the tag.
+strip_branch_tag_prefix() {
+    tag="${1}"
+    current_branch="${2}"
+
+    echo ${tag#${current_branch}-}
+}
+
+
+current_branch=$(git -C ${SNAPCRAFT_PART_SRC} branch --show-current)
+
+# If current commit on that branch is tagged, this is the version we are using.
+current_commit_tag=$(git -C ${SNAPCRAFT_PART_SRC} describe --exact-match --abbrev=0 --tags 2>/dev/null) || true
+if [ -n "${current_commit_tag}" ]; then
```

Comment:
Could be simplified to 
```suggestion
if ! git -C "${SNAPCRAFT_PART_SRC}" describe --exact-match --abbrev=0 --tags 2>/dev/null; then
```

Reply:
Ah nevermind, we're using the `$current_commit_tag` value below

Reply:
Exactly!